File: Java/KodeProgram/Produk.java
// Produk.java
// Berisi definisi class dasar abstrak Produk.
abstract class Produk {
    private int id;
    private String nama;
    private double harga;

    public Produk(int id, String nama, double harga) {
        this.id = id;
        this.nama = nama;
        this.harga = harga;
    }

    public int getId() { return id; }
    public String getNama() { return nama; }
    public double getHarga() { return harga; }

    public abstract String getKategori();
    public abstract String getDetail();
}

File: PHP/KodeProgram/Produk.php
<?php
// Produk.php
abstract class Produk {
    private $id;
    private $nama;
    private $harga;

    public function __construct($id, $nama, $harga) {
        $this->id = $id;
        $this->nama = $nama;
        $this->harga = $harga;
    }

    public function getId() {
        return $this->id;
    }

    public function getNama() {
        return $this->nama;
    }

    public function getHarga() {
        return $this->harga;
    }

    abstract public function getKategori();
    abstract public function getDetail();
}